<a href="https://colab.research.google.com/github/Philemonkipkirui/learningGit/blob/master/Week1_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [74]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [75]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [76]:
df = pd.read_csv('/content/drive/MyDrive/1. Weather Data.csv')

In [77]:
df.head(10)

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0  1/1/2012 0:00    -1.8              -3.9         86                4   
1  1/1/2012 1:00    -1.8              -3.7         87                4   
2  1/1/2012 2:00    -1.8              -3.4         89                7   
3  1/1/2012 3:00    -1.5              -3.2         88                6   
4  1/1/2012 4:00    -1.5              -3.3         88                7   
5  1/1/2012 5:00    -1.4              -3.3         87                9   
6  1/1/2012 6:00    -1.5              -3.1         89                7   
7  1/1/2012 7:00    -1.4              -3.6         85                7   
8  1/1/2012 8:00    -1.4              -3.6         85                9   
9  1/1/2012 9:00    -1.3              -3.1         88               15   

   Visibility_km  Press_kPa               Weather  
0            8.0     101.24                   Fog  
1            8.0     101.24                   Fog  
2            4.0     101.26  Freezing Drizzle,Fog  
3            4.0     101.27  Freezing Drizzle,Fog  
4            4.8     101.23                   Fog  
5            6.4     101.27                   Fog  
6            6.4     101.29                   Fog  
7            8.0     101.26                   Fog  
8            8.0     101.23                   Fog  
9            4.0     101.20                   Fog

Weather Being Clear.

In [78]:
clear_weather = df.query('Weather == "Clear"')
#clear_weather = df[df["Weather"]=='Clear']
clear_weather

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
67      1/3/2012 19:00   -16.9             -24.8         50               24   
114     1/5/2012 18:00    -7.1             -14.4         56               11   
115     1/5/2012 19:00    -9.2             -15.4         61                7   
116     1/5/2012 20:00    -9.8             -15.7         62                9   
117     1/5/2012 21:00    -9.0             -14.8         63               13   
...                ...     ...               ...        ...              ...   
8646   12/26/2012 6:00   -13.4             -14.8         89                4   
8698  12/28/2012 10:00    -6.1              -8.6         82               19   
8713   12/29/2012 1:00   -11.9             -13.6         87               11   
8714   12/29/2012 2:00   -11.8             -13.1         90               13   
8756  12/30/2012 20:00   -13.8             -16.5         80               24   

      Visibility_km  Press_kPa Weather  
67             25.0     101.74   Clear  
114            25.0     100.71   Clear  
115            25.0     100.80   Clear  
116            25.0     100.83   Clear  
117            25.0     100.83   Clear  
...             ...        ...     ...  
8646           25.0     102.47   Clear  
8698           24.1     101.27   Clear  
8713           25.0     101.31   Clear  
8714           25.0     101.33   Clear  
8756           25.0     101.52   Clear  

[1326 rows x 8 columns]

Wind speed being exactly 4km/h


In [79]:
df.rename(columns={"Wind Speed_km/h":"Wind_Speed_km_h", }, inplace= True)
df.columns
Wind_Speed_4km = df.query('Wind_Speed_km_h == 4').value_counts()
#Wind_Speed_4km_counts = df['Wind_Speed_km_h'].value_counts()
Wind_Speed_4km
#Wind_Speed_4km_counts

Date/Time        Temp_C  Dew Point Temp_C  Rel Hum_%  Wind_Speed_km_h  Visibility_km  Press_kPa  Weather      
1/1/2012 0:00    -1.8    -3.9              86         4                8.0            101.24     Fog              1
5/23/2012 4:00    12.6    9.4              81         4                24.1           100.87     Mostly Cloudy    1
5/5/2012 20:00    12.8   -2.0              36         4                25.0           101.62     Mainly Clear     1
5/5/2012 0:00     8.4     4.1              74         4                25.0           101.25     Mainly Clear     1
5/4/2012 18:00    17.4    13.0             75         4                24.1           100.79     Mostly Cloudy    1
                                                                                                                 ..
12/26/2012 8:00  -13.7   -15.2             88         4                16.1           102.53     Mostly Cloudy    1
12/26/2012 6:00  -13.4   -14.8             89         4                25.0           102.47     Clear            1
12/26/2012 5:00  -12.7   -14.1             89         4                25.0           102.48     Clear            1
12/26/2012 2:00  -12.7   -14.4             87         4                25.0           102.45     Clear            1
9/7/2012 20:00    23.2    11.9             49         4                25.0           100.58     Mostly Cloudy    1
Name: count, Length: 474, dtype: int64

474 Times where the wind speed was 4km/h

Checking for Nulls

In [80]:
null_count = df.isnull().sum()
null_count

Date/Time           0
Temp_C              0
Dew Point Temp_C    0
Rel Hum_%           0
Wind_Speed_km_h     0
Visibility_km       0
Press_kPa           0
Weather             0
dtype: int64

The are no nulll values in the data frame

Renaming of the "Weather" column to Waether_Conditions

In [81]:
Weather_column_rename = df.rename(columns={"Weather" : "Weather_Condition"}, inplace = True)
df.columns

Index(['Date/Time', 'Temp_C', 'Dew Point Temp_C', 'Rel Hum_%',
       'Wind_Speed_km_h', 'Visibility_km', 'Press_kPa', 'Weather_Condition'],
      dtype='object')

Mean Visibility of the Data Set.

In [82]:
Visibility_mean= df['Visibility_km'] .mean()
Visibility_mean

27.664446721311478

Wind Speed being greater than 24km/h and Visibility being 25km

In [83]:
test_1_df = df.query('Wind_Speed_km_h > 24 and Visibility_km == 25').value_counts()
test_1_df

Date/Time        Temp_C  Dew Point Temp_C  Rel Hum_%  Wind_Speed_km_h  Visibility_km  Press_kPa  Weather_Condition
1/1/2012 23:00    5.3     2.0              79         30               25.0           99.31      Cloudy               1
2/26/2012 4:00   -10.8   -15.3             69         30               25.0           101.60     Mostly Cloudy        1
2/5/2012 23:00   -1.5    -5.7              73         30               25.0           100.93     Cloudy               1
2/5/2012 22:00   -2.1    -6.7              71         28               25.0           101.06     Cloudy               1
2/5/2012 21:00   -2.5    -7.7              67         26               25.0           101.18     Cloudy               1
                                                                                                                     ..
10/30/2012 5:00   13.7    10.5             81         37               25.0           99.32      Mainly Clear         1
10/30/2012 4:00   13.8    10.4             80         35               25.0           99.28      Mostly Cloudy        1
10/30/2012 3:00   13.9    10.6             80         41               25.0           99.21      Rain Showers         1
10/30/2012 2:00   14.1    10.3             78         37               25.0           99.31      Mostly Cloudy        1
9/9/2012 1:00     13.9    9.6              75         28               25.0           100.26     Cloudy               1
Name: count, Length: 308, dtype: int64

384 data points where the wind speed is greater than 24km/h and visibility being 25km in the data frame

Determining the Mean for each column with integer values.

In [84]:
df_mean = df.select_dtypes(include='number').mean()
df_mean

Temp_C                8.798144
Dew Point Temp_C      2.555294
Rel Hum_%            67.431694
Wind_Speed_km_h      14.945469
Visibility_km        27.664447
Press_kPa           101.051623
dtype: float64

Determining instances where weather is clear and relative humidity is 50% or visibility being greater than 40km

In [86]:
test_rename = df.rename(columns={"Rel Hum_%": "Rel_Hum"}, inplace = True)
test_2_df = df.query('Weather_Condition	== "Clear" and Rel_Hum == 50 or Visibility_km > 40').value_counts()
test_2_df

Date/Time        Temp_C  Dew Point Temp_C  Rel_Hum  Wind_Speed_km_h  Visibility_km  Press_kPa  Weather_Condition
1/14/2012 12:00  -17.5   -23.8             58       20               48.3           101.16     Mostly Cloudy        1
7/11/2012 5:00    13.1    8.7              75       4                48.3           101.66     Clear                1
7/12/2012 18:00   29.1    13.6             39       13               48.3           101.32     Cloudy               1
7/12/2012 17:00   29.8    13.1             36       19               48.3           101.33     Mostly Cloudy        1
7/12/2012 16:00   30.4    13.0             34       24               48.3           101.37     Mostly Cloudy        1
                                                                                                                   ..
4/18/2012 18:00   10.0   -5.5              33       7                48.3           101.70     Mainly Clear         1
4/18/2012 17:00   10.2   -4.1              36       13               48.3           101.75     Mainly Clear         1
4/18/2012 16:00   10.3   -3.6              37       9                48.3           101.80     Mainly Clear         1
4/18/2012 15:00   10.4   -3.2              38       11               48.3           101.86     Mainly Clear         1
9/9/2012 9:00     14.8    8.8              67       17               48.3           100.65     Mainly Clear         1
Name: count, Length: 2026, dtype: int64

There 2026 points where weather is clear and relative humidity is 50% or visibility being greater than 40km

The Number of weather conditions that include Snow

In [87]:
weather_snow = df[df['Weather_Condition'].str.contains('Snow')]
weather_snow

Date/Time  Temp_C  Dew Point Temp_C  Rel_Hum  Wind_Speed_km_h  \
41      1/2/2012 17:00    -2.1              -9.5       57               22   
44      1/2/2012 20:00    -5.6             -13.4       54               24   
45      1/2/2012 21:00    -5.8             -12.8       58               26   
47      1/2/2012 23:00    -7.4             -14.1       59               17   
48       1/3/2012 0:00    -9.0             -16.0       57               28   
...                ...     ...               ...      ...              ...   
8779  12/31/2012 19:00     0.1              -2.7       81               30   
8780  12/31/2012 20:00     0.2              -2.4       83               24   
8781  12/31/2012 21:00    -0.5              -1.5       93               28   
8782  12/31/2012 22:00    -0.2              -1.8       89               28   
8783  12/31/2012 23:00     0.0              -2.1       86               30   

      Visibility_km  Press_kPa Weather_Condition  
41             25.0      99.66      Snow Showers  
44             25.0     100.07      Snow Showers  
45             25.0     100.15      Snow Showers  
47             19.3     100.27      Snow Showers  
48             25.0     100.35      Snow Showers  
...             ...        ...               ...  
8779            9.7     100.13              Snow  
8780            9.7     100.03              Snow  
8781            4.8      99.95              Snow  
8782            9.7      99.91              Snow  
8783           11.3      99.89              Snow  

[583 rows x 8 columns]

The Number of weather conditions that include Snow is 583